### PR TITLE
chore: add docs for adoption

### PR DESCRIPTION
**Pull request by davidjgoss**
### 🤔 What's changed?

Add a section to the README about how to adopt the compatibility kit in your Cucumber implementation. Move the contributing docs to the standard file for clarity.

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Contributing
+
+This repo is composed of a [devkit](./devkit/), and packages to deliver the kit
+using managers like NPM or rubygem.
```
Comment (luke-hill) [inline CONTRIBUTING.md:4]: rubygems.

File: README.md [hunk 2]
```diff
@@ -12,33 +12,49 @@ features from the kit.
...
 The kit is composed of features, step definitions, and messages:
 
-- features, once executed, emit an exhaustive set of Messages as specified by
-  the protocol
-- step definitions allows to execute the features of the kit as expected
-- Messages - serialized as `.ndjson` files - are the reference: a given feature
+- **features**, once executed, emit an exhaustive set of Messages as specified by the protocol
+- **step definitions** that are used in conjunction with the features to generate the Messages, and also serve as reference step definitions for implementations adopting the kit
+- **Messages**, serialized as `.ndjson` files, are the reference: a given feature
   from the kit, executed using its dedicated step definitions, must emit the
```
Comment (luke-hill) [inline README.md:18]: probably bold **must**

File: README.md [hunk 3]
```diff
@@ -12,33 +12,49 @@ features from the kit.
...
 The Messages from the kit are generated using
 [fake-cucumber](https://github.com/cucumber/fake-cucumber), which is used here
 as a reference implementation of the Messages protocol.
 
-## Contributing
+## Adoption
 
-This repo is composed of a [devkit](./devkit/), and packages to deliver the kit
-using managers like NPM or rubygem.
+To adopt the compatibility kit when writing a Cucumber implementation:
```
Comment (luke-hill) [inline README.md:27]: implementation, a cucumber implementer must ...

File: README.md [hunk 4]
```diff
@@ -12,33 +12,49 @@ features from the kit.
...
 
-### The devkit
+### Prerequisites
 
-The [devkit](./devkit/) is a set of tools to generate new Messages in order to
-extend the kit, or to update it after the protocol has been changed.
+First, make sure your implementation is sufficiently testable - you should be able to execute a test run programatically for a given set of feature files and step definitions, and get a handle on the Messages output (via a formatter or some other means) to make assertions on.
```
Comment (luke-hill) [inline README.md:31]: and obtain the Messages output (in .ndjson format, via a formatter or some other means), to make assertions on.

File: README.md [hunk 5]
```diff
@@ -12,33 +12,49 @@ features from the kit.
...
+
+### Notes and exceptions
+
+When we say "matches" above, that's heavily qualified - there are many individual fields that will vary from run to run, like generated identifiers, timestamps, durations and file URLs. You can freely exclude these fields from your assertion - the key thing to capture is the number and order of the messages, and the content that would be consistent across runs. It's also not unheard of for implementations to fix up the order of messages so it matches the CCK, although it's best to try to match it naturally if you can.
+
+A small minority of features also have an `.arguments.txt` that specify additional arguments/options that are pertinent to the feature - see `retry` for an example. You can adapt these how you see fit - they may or may not line up directly with your CLI options schema.
+
+There may be some features in the CCK suite that cover functionality that your implementation doesn't have. For example, there's one for `retry` which only a subset of Cucumbers support. You can filter these out of your test until/unless you're ready to support them.
```
Comment (luke-hill) [inline README.md:55]: only a subset of Cucumber implementations support.

**Review by luke-hill — COMMENTED** [review]
few minor language tweaks

**Review by luke-hill — APPROVED** [review]
Get some other views too. But sounds good so far.

File: README.md [hunk 6]
```diff
@@ -12,33 +12,49 @@ features from the kit.
...
 The Messages from the kit are generated using
 [fake-cucumber](https://github.com/cucumber/fake-cucumber), which is used here
 as a reference implementation of the Messages protocol.
 
-## Contributing
+## Adoption
 
-This repo is composed of a [devkit](./devkit/), and packages to deliver the kit
-using managers like NPM or rubygem.
+To adopt the compatibility kit when writing a Cucumber implementation, an implementer must:
```
Comment (mattwynne) [inline README.md:27]: This bit reads a little odd when you look at it in the rendered document, as what follows the colon is a heading.

I'd probably summarize it here (e.g. "an implementor must first satisfy some prerequisites, then ...")

File: README.md [hunk 7]
```diff
@@ -12,33 +12,49 @@ features from the kit.
...
-  the CCK.
+Next, for each feature in the CCK suite, write step definitions that work with your implementation - these should be _equivalent_ to the reference step definitions from the devkit, but with the appropriate language, syntax etc.
+
+### Testing
+
+Now you can build your CCK test. For each feature in the CCK suite, it should:
+
+1. Execute a test run, scoped to:
+  - Only the feature file in question
+  - Only the step definition file that you wrote for this feature
```
Comment (luke-hill) [inline README.md:45]: Should we add an NB here. The CCK will **not** tell you if the step definition is correctly scoping the cuke in question? As we have found out :D

File: README.md [hunk 8]
```diff
@@ -12,33 +12,49 @@ features from the kit.
...
+2. Capture the Messages output of that test run
+3. Load the reference messages from the `.ndjson` file and assert that the output you captured matches it
+
+### Notes and exceptions
+
+When we say "matches" above, that's heavily qualified - there are many individual fields that will vary from run to run, like generated identifiers, timestamps, durations and file URLs. You can freely exclude these fields from your assertion - the key thing to capture is the number and order of the messages, and the content that would be consistent across runs. It's also not unheard of for implementations to fix up the order of messages so it matches the CCK, although it's best to try to match it naturally if you can.
```
Comment (luke-hill) [inline README.md:51]: type, number and order of the messages - again as we've found out :D type is sometimes not correct haha

**Review by ehuelsmann — COMMENTED** [review]
Thanks! This definitely will help me get started.

**Review by ehuelsmann — APPROVED** [review]

File: README.md (hunk 8 above)
Comment (davidjgoss) [reply README.md:51]: Done

File: README.md (hunk 7 above)
Comment (davidjgoss) [reply README.md:45]: Good point, added a paragraph for that.

File: README.md (hunk 6 above)
Comment (davidjgoss) [reply README.md]: Fixed

**Pull request merged**